File: apps/web/src/api/emuShiftReportContracts.ts
export type EmuShiftType = "day" | "night";
export type EmuShiftReportCategory = "mechanic" | "electrician";

export interface EmuShiftReportEmployeeOptionDto { id: string; fullName: string; personnelNo: string; position: string; department: string; workerCategory: EmuShiftReportCategory; }
export interface EmuShiftReportSectionDto { id: string; name: string; code: string; isActive: boolean; sortOrder: number; }
export interface EmuShiftReportShiftOptionDto { shiftType: EmuShiftType; name: string; startTime: string; endTime: string; crossesMidnight: boolean; }
export interface EmuShiftReportOptionsDto { employees: EmuShiftReportEmployeeOptionDto[]; sections: EmuShiftReportSectionDto[]; shifts: EmuShiftReportShiftOptionDto[]; }
export interface EmuCreateShiftReportLineDto { workDescription: string; durationMinutes: number; sectionId?: string | null; note?: string | null; }
export interface EmuCreateShiftReportDto { reportDate: string; shiftType: EmuShiftType; workerCategory: EmuShiftReportCategory; employeeId: string; lines: EmuCreateShiftReportLineDto[]; }
export interface EmuShiftReportLineDto extends EmuCreateShiftReportLineDto { id: string; sequenceNo: number; sectionName: string; }
export interface EmuShiftReportSummaryDto { id: string; reportDate: string; shiftType: EmuShiftType; workerCategory: EmuShiftReportCategory; employeeId: string; employeeName: string; personnelNo: string; position: string; department: string; status: string; workCount: number; totalDurationMinutes: number; createdByUserId: string | null; createdByName: string; submittedAt: string; }
export interface EmuShiftReportDetailDto extends EmuShiftReportSummaryDto { lines: EmuShiftReportLineDto[]; }
export interface EmuShiftReportListResponseDto { rows: EmuShiftReportSummaryDto[]; total: number; page: number; pageSize: number; pageCount: number; }
export interface EmuShiftReportQuery { date?: string; shiftType?: EmuShiftType | ""; workerCategory?: EmuShiftReportCategory | ""; employeeId?: string; search?: string; page?: number; pageSize?: number; }

File: libs/contracts/EmuShiftReportContracts.cs
namespace Patrol360.Contracts;

public sealed record EmuShiftReportEmployeeOptionDto(Guid Id, string FullName, string PersonnelNo, string Position, string Department, string? WorkerCategory, string? AssignedWorkerCategory);
public sealed record EmuSetShiftReportEmployeeCategoryDto(string? WorkerCategory);
public sealed record EmuShiftReportShiftOptionDto(string ShiftType, string Name, TimeOnly StartTime, TimeOnly EndTime, bool CrossesMidnight);
public sealed record EmuShiftReportOptionsDto(IReadOnlyList<EmuShiftReportEmployeeOptionDto> Employees, IReadOnlyList<EmuReferenceDto> Sections, IReadOnlyList<EmuShiftReportShiftOptionDto> Shifts);
public sealed record EmuCreateShiftReportLineDto(string WorkDescription, int DurationMinutes, Guid? SectionId, string? Note);
public sealed record EmuCreateShiftReportDto(DateOnly ReportDate, string ShiftType, string WorkerCategory, Guid EmployeeId, IReadOnlyList<EmuCreateShiftReportLineDto> Lines);
public sealed record EmuShiftReportLineDto(Guid Id, int SequenceNo, string WorkDescription, int DurationMinutes, Guid? SectionId, string SectionName, string Note);
public sealed record EmuShiftReportSummaryDto(Guid Id, DateOnly ReportDate, string ShiftType, string WorkerCategory, Guid EmployeeId, string EmployeeName, string PersonnelNo, string Position, string Department, string Status, int WorkCount, int TotalDurationMinutes, Guid? CreatedByUserId, string CreatedByName, DateTimeOffset SubmittedAt);
public sealed record EmuShiftReportDetailDto(Guid Id, DateOnly ReportDate, string ShiftType, string WorkerCategory, Guid EmployeeId, string EmployeeName, string PersonnelNo, string Position, string Department, string Status, int WorkCount, int TotalDurationMinutes, Guid? CreatedByUserId, string CreatedByName, DateTimeOffset SubmittedAt, IReadOnlyList<EmuShiftReportLineDto> Lines);
public sealed record EmuShiftReportQueryDto(DateOnly? Date = null, DateOnly? DateFrom = null, DateOnly? DateTo = null, string? ShiftType = null, string? WorkerCategory = null, Guid? EmployeeId = null, string? Search = null, int Page = 1, int PageSize = 50, bool FavoriteOnly = false);
